api/objects.proto

```diff
@@ -48,6 +48,13 @@ message Node {
 
 	// Certificate is the TLS certificate issued for the node, if any.
 	Certificate certificate = 8 [(gogoproto.nullable) = false];
+
+	// Role is the *observed* role for this node. It differs from the
+	// desired role set in Node.Spec.Role because the role here is only
+	// updated after the Raft member list has been reconciled with the
+	// desired role from the spec. Note that this doesn't show whether the
+	// node has obtained a certificate that reflects its current role.
+	NodeRole role = 9;
```

Comment:
I agree that your suggestion of `reconciled_role` would be more clear here, since you're right that "observed" implies (to me at least) the role of the certificate the node is currently using.